### PR TITLE
Replace `github.com/ghodss/yaml` with `sigs.k8s.io/yaml`

### PR DESCRIPTION
**Pull request by Juneezee**
## Description

The `github.com/ghodss/yaml` package is no longer actively maintained. There are numerous inquiries about the project's status on its issue tracker: https://github.com/ghodss/yaml/issues. `sigs.k8s.io/yaml` is a permanent fork of `github.com/ghodss/yaml`, which is actively maintained by Kubernetes SIG and widely used in K8s projects.

Since `sigs.k8s.io/yaml` was already an indirect dependency before this pull request was made, we can remove 1 extra dependency by replacing `github.com/ghodss/yaml` with `sigs.k8s.io/yaml`.

<details><summary>Output of <code>make ut</code> for <code>pkg/crds</code></summary>

```
Running Suite: pkg/crds Suite
=============================
Random Seed: 1689946969
Will run 5 of 5 specs

test crds pkg GetCalicoCRDSource 
  should quickly load calico source CRDs
  /go/src/github.com/tigera/operator/pkg/crds/crds_test.go:26

• [MEASUREMENT]
test crds pkg
/go/src/github.com/tigera/operator/pkg/crds/crds_test.go:24
  GetCalicoCRDSource
  /go/src/github.com/tigera/operator/pkg/crds/crds_test.go:25
    should quickly load calico source CRDs
    /go/src/github.com/tigera/operator/pkg/crds/crds_test.go:26

    Ran 50 samples:
    runtime:
      Fastest Time: 0.000s
      Slowest Time: 0.002s
      Average Time: 0.001s ± 0.000s
------------------------------
test crds pkg GetEnterpriseCRDSource 
  should quickly load enterprise source CRDs
  /go/src/github.com/tigera/operator/pkg/crds/crds_test.go:34

• [MEASUREMENT]
test crds pkg
/go/src/github.com/tigera/operator/pkg/crds/crds_test.go:24
  GetEnterpriseCRDSource
  /go/src/github.com/tigera/operator/pkg/crds/crds_test.go:33
    should quickly load enterprise source CRDs
    /go/src/github.com/tigera/operator/pkg/crds/crds_test.go:34

    Ran 50 samples:
    runtime:
      Fastest Time: 0.001s
      Slowest Time: 0.002s
      Average Time: 0.001s ± 0.000s
------------------------------
test crds pkg GetOperatorCRDSource 
  should quickly load operator source CRDs
  /go/src/github.com/tigera/operator/pkg/crds/crds_test.go:42

• [MEASUREMENT]
test crds pkg
/go/src/github.com/tigera/operator/pkg/crds/crds_test.go:24
  GetOperatorCRDSource
  /go/src/github.com/tigera/operator/pkg/crds/crds_test.go:41
    should quickly load operator source CRDs
    /go/src/github.com/tigera/operator/pkg/crds/crds_test.go:42

    Ran 50 samples:
    runtime:
      Fastest Time: 0.001s
      Slowest Time: 0.002s
      Average Time: 0.001s ± 0.000s
------------------------------
test crds pkg GetCRDs 
  should quickly load calico CRDs
  /go/src/github.com/tigera/operator/pkg/crds/crds_test.go:50

• [MEASUREMENT]
test crds pkg
/go/src/github.com/tigera/operator/pkg/crds/crds_test.go:24
  GetCRDs
  /go/src/github.com/tigera/operator/pkg/crds/crds_test.go:49
    should quickly load calico CRDs
    /go/src/github.com/tigera/operator/pkg/crds/crds_test.go:50

    Ran 50 samples:
    runtime:
      Fastest Time: 0.001s
      Slowest Time: 0.162s
      Average Time: 0.007s ± 0.022s
------------------------------
test crds pkg GetCRDs 
  should quickly load enterprise CRDs
  /go/src/github.com/tigera/operator/pkg/crds/crds_test.go:56

• [MEASUREMENT]
test crds pkg
/go/src/github.com/tigera/operator/pkg/crds/crds_test.go:24
  GetCRDs
  /go/src/github.com/tigera/operator/pkg/crds/crds_test.go:49
    should quickly load enterprise CRDs
    /go/src/github.com/tigera/operator/pkg/crds/crds_test.go:56

    Ran 50 samples:
    runtime:
      Fastest Time: 0.004s
      Slowest Time: 0.199s
      Average Time: 0.009s ± 0.027s
------------------------------

JUnit report was created: /go/src/github.com/tigera/operator/report/ut/crds_suite.xml

Ran 5 of 5 Specs in 0.952 seconds
SUCCESS! -- 5 Passed | 0 Failed | 0 Pending | 0 Skipped
PASS
```

</details> 

## For PR author

- [x] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

**Comment by rene-dekker** [comment]
Thank you for your contribution. It looks like the CI failed due to a file copyright being outdated. Please change
`pkg/crds/crds.go:1` to `2021-2023`.

**Review by rene-dekker — APPROVED** [review]
lgtm

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and delete the branch after I've merged it.

**Comment by marvin-tigera** [comment]
I'm sorry but I failed delete the branch after merging the pull request.

**Pull request merged**
